src/core/boxes.c

```diff
@@ -577,7 +577,6 @@ meta_rectangle_get_minimal_spanning_set_for_region (
   *temp_rect = *basic_rect;
   ret = g_list_prepend (NULL, temp_rect);
 
-  strut_iter = all_struts;
```

Comment:
This most likely is optimized out by compiler... So I doubt it improves anything.

Reply:
source code can not depend on binary optimization done by the compiler for unfound sentences like this.

Reply:
I am not saying that you should relay that compiler will optimize it out. Of course this is obviously unneeded assignment and can be removed.

You are claiming to improve performance, where are numbers? And under what test conditions?

Any sane compiler will optimize out this redundant assignment so unless you disable optimizations this improves performance by exactly 0%. `-01` with gcc will be enough to optimize out redundant assignments or completely remove unused variables. So in any real life usage there wont be any performance improvement.

And even if optimizations are disabled what are you gaining? Save 1 cpu cycle (of course with modern cpus that is more complicated)?